File: 스킬트리.py
def solution(skill, skill_trees):
    skill_book=[]
    temp = []
    answer = 0
    
    for s in range(len(skill)) :        #선행 스킬 정리
        skill_book.append(skill[s:s+1])
            
    for st in skill_trees :  #선행 스킬을 스킬 트리에서의 위치 저장
        for i in range(len(skill_book)) :  
            if st.find(skill_book[i]) == -1 : temp.append(27)
            else : temp.append(st.find(skill_book[i]))


        print(temp)
        c = 0
        
        for j in range(len(temp)-1) :                
            if temp[j] > temp[j+1] :
                c = 1
                break                
        print(c)
        if c == 0 :
            answer += 1
        temp.clear()

    return answer

File: 프로그래머스/lv2/49993. 스킬트리/스킬트리.js
function solution(skill, skill_trees) {
    var answer = 0; 
    
    skill_trees.forEach((tree)=>{
        let count = 0;
        let possible = true;
        
        for(let i = 0; i <tree.length; i++){
            console.log(tree[i]);
            if(skill.includes(tree[i])){
                if(skill.indexOf(tree[i]) === count){
                    count++;
                }else{
                    possible = false;
                    break;
                }
            }
        }
        
        if(possible){
            answer++;
        }
        
    })
    return answer;
}